proj/views.py

```diff
@@ -286,88 +244,302 @@ def _cart_id(request):
+            messages.info(request, "This item quantity was updated.")
+            return redirect('cart_page')
+        else:
+            messages.info(request, "This item was not in your cart")
+            return redirect('product_detail', slug=slug)
+    else:
+        messages.info(request, "You do not have an active order")
+        return redirect('product_detail', slug=slug)
 
-class CartPage(DetailView):
-    template_name = 'cart_page.html'
-    model = CartItemm
 
-    def get_object(self, queryset=None):
-        username = self.kwargs['username']
-        user = get_object_or_404(User, username=username)
-        return user
+class CartPage(LoginRequiredMixin, View):
+    def get(self, *args, **kwargs):
+        try:
+            order = Order.objects.get(user=self.request.user, ordered=False)
 
-    def get_queryset(self):
-        user = self.get_object()
-        queryset = CartItemm.objects.filter(user=user)
-        return queryset
+            context = {
+                'object': order,
+            }
+            return render(self.request, 'cart_page.html', context)
+        except ObjectDoesNotExist:
+            messages.warning(self.request, "You do not have an active order")
+            return redirect("/")
 
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        context['carts'] = self.get_queryset()
-        cart_items = self.get_queryset()
-        cart_total = sum(cart_item.total() for cart_item in cart_items)
-        context['cart_total'] = cart_total
-        return context
 
-
-def add_compare_product(request, pk):
-    product = get_object_or_404(Product, id=pk)
+class CheckoutView(View):
+    def get(self, *args, **kwargs):
+        try:
+            order = Order.objects.get(user=self.request.user, ordered=False)
+            form = CheckoutForm()
+            context = {
+                'form': form,
+                'couponform': CouponForm(),
+                'order': order,
+                'DISPLAY_COUPON_FORM': True
+            }
+
+            shipping_address_qs = Address.objects.filter(
+                user=self.request.user,
+                address_type='S',
+                default=True
+            )
+            if shipping_address_qs.exists():
+                context.update(
+                    {'default_shipping_address': shipping_address_qs[0]})
+
+            billing_address_qs = Address.objects.filter(
+                user=self.request.user,
+                address_type='B',
+                default=True
+            )
+            if billing_address_qs.exists():
+                context.update(
+                    {'default_billing_address': billing_address_qs[0]})
+            return render(self.request, "checkout_page.html", context)
+        except ObjectDoesNotExist:
+            messages.info(self.request, "You do not have an active order")
+            return redirect("checkout")
+
+    def post(self, *args, **kwargs):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 72 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.